remotes/docker/fetcher.go

```diff
@@ -82,6 +82,10 @@ func (r dockerFetcher) Fetch(ctx context.Context, desc ocispec.Descriptor) (io.R
 					continue // try one of the other urls.
 				}
 
+				if host.Scheme != "http" && host.Scheme != "https" {
```

Comment:
You can ignore the error regardless to the scheme.
The spec says `This OPTIONAL property specifies a list of URIs from which this object MAY be downloaded. `. There is no guarantee that the object is available at the http(s) URL.

https://github.com/opencontainers/image-spec/blob/v1.0/descriptor.md#properties

Reply:
I agree with the reading of the spec, but I'm not sure that this is the right end result.

A pre-existing user of containerd's resolver might depend on knowing an HTTP transaction errored out for some valid use of http(s) URLs in current images. Now they get a silent failure (with log entry) and can't handle that failure programmatically. Am I missing anything?

Reply:
Another possible way we can take is filtering `urls` to only `http(s)://` URLs and calling `open()` only for them.

cc @AkihiroSuda @dmcgowan

Reply:
> Another possible way we can take is filtering `urls` to only `http(s)://` URLs and calling `open()` only for them.
> 
> cc @AkihiroSuda @dmcgowan

SGTM